sky/resources.py

```diff
@@ -305,6 +314,7 @@ def memory(self) -> Optional[str]:
         return self._memory
 
     @property
+    @functools.lru_cache()
```

Comment:
Why do we need this?

Reply:
This is to reduce the frequency we call into the `service_catalog` underneath to reduce the overhead, i.e., searching the df for the accelerators. Since the `resources` object is immutable, caching this function should be safe.

Reply:
- Is the overhead noticeable? Worth documenting if so.
- Shall we add `max_size=1`?

Reply:
The overhead is not significant, but it seems no harm to cache the output. The following commands take 3.3 seconds in master branch, but 0.008 seconds in the current PR.
```
python -u <<EOF                            
import sky
import time
r = sky.Resources(instance_type='p3.2xlarge')
total_time = 0
for i in range(1000): 
  start = time.time()
  str(r)
  total_time += time.time() - start
print(total_time)

EOF
```